### PR TITLE
Add snippets for kotlin

### PR DESCRIPTION
**Pull request by jojojames**

### CODE REVIEW

File: snippets/kotlin-mode/maino [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- mode: snippet -*-
+# name: JvmStatic main function
+# key: maino
+# --
```
Comment (AndreaCrotti) [inline snippets/kotlin-mode/maino]: uhm not sure it make sense to have a different snippet which is the same as `main` except for the `@JvmStatic` on top.
Is it something that it's so commonly used?
And what does the `o` in `maino` is for anyway?

File: snippets/kotlin-mode/fun1 [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# -*- mode: snippet -*-
+# name: Function with one parameter
+# key: fun1
+# --
+fun $1($2: $3): $4 {
```
Comment (AndreaCrotti) [inline snippets/kotlin-mode/fun1]: I would just have a generic fun where you can step into the parameters and declare as many as you want.
`fun $1(${2:parameters}): $3..` or something like that.
Would not add a different snippet for each possible number of arguments (even if I agree that possibly 1-2 arguments are quite common cases, but it's not something done for other snippets in this repository at least)

File: snippets/kotlin-mode/maino (hunk 1 above)
Comment (jojojames) [reply snippets/kotlin-mode/maino]: These are from https://github.com/JetBrains/kotlin/blob/master/idea/resources/liveTemplates/Kotlin.xml

I'm not sure how common it is but they were in ^ so I included them.

File: snippets/kotlin-mode/fun1 (hunk 2 above)
Comment (jojojames) [reply snippets/kotlin-mode/fun1]: Same as above^, I can remove them and use your suggestion if you think that's better.

One value add of these multi-parameter snippets is these functions expand to

fun functionName(parameterOne: parameterOneType).

Comment (AndreaCrotti) [reply snippets/kotlin-mode/fun1]: Mm yes I would remove these two fun1 and fun2, and just make fun a bit more generic to just define as many parameters as you like..

File: snippets/kotlin-mode/maino (hunk 1 above)
Comment (AndreaCrotti) [reply snippets/kotlin-mode/maino]: well I would remove `maino` as well, doesn't make much sense to me to have this kind of snippet that simply annotates another snippet..

File: snippets/kotlin-mode/fun1 (hunk 2 above)
Comment (jojojames) [reply snippets/kotlin-mode/fun1]: Done.

File: snippets/kotlin-mode/maino (hunk 1 above)
Comment (jojojames) [reply snippets/kotlin-mode/maino]: Done.

**Comment by AndreaCrotti** [comment]
Thanks @jojojames , merging

**Pull request merged**